### PR TITLE
Add integration test

### PR DESCRIPTION
**Pull request by kerthcet**
#### What type of PR is this?
/kind feature

#### Which issue(s) this PR fixes

Fixes #

#### Special notes for your reviewer

Will add presubmit to test-infra as well.

#### Does this PR introduce a user-facing change?

```release-note
Run `make test-integration` to run integration tests locally.
```

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -72,6 +73,12 @@ vet: ## Run go vet against code.
 test: manifests fmt vet envtest ## Run tests.
 	KUBEBUILDER_ASSETS="$(shell $(ENVTEST) use $(ENVTEST_K8S_VERSION) --bin-dir $(LOCALBIN) -p path)" go test ./... -coverprofile cover.out
 
+.PHONY: integration-test
+integration-test: manifests fmt vet envtest ginkgo ## Run integration tests.
```
Comment (ahg-g) [inline Makefile:77]: just so we are consistent with the other projects, can we flip this to test-integration please ?

File: Makefile [hunk 2]
```diff
@@ -72,6 +73,12 @@ vet: ## Run go vet against code.
 test: manifests fmt vet envtest ## Run tests.
 	KUBEBUILDER_ASSETS="$(shell $(ENVTEST) use $(ENVTEST_K8S_VERSION) --bin-dir $(LOCALBIN) -p path)" go test ./... -coverprofile cover.out
```
Comment (ahg-g) [inline Makefile:74]: should we change this to use gotestsum?

https://github.com/kubernetes-sigs/jobset/blob/main/Makefile#L123-L125

Comment (kerthcet) [reply Makefile]: Make sense, done.

File: Makefile (hunk 1 above)
Comment (kerthcet) [reply Makefile]: Done

**Comment by ahg-g** [comment]
/label tide/merge-method-squash

/lgtm
/approve
/hold

until the other pr merges

**Comment by kerthcet** [comment]
Replace test with `./...` with `./api/... ./pkg/...` or the former will run the integration tests as well.

**Comment by ahg-g** [comment]
it is strange that we still have pull-lws-test-main, which got renamed to pull-lws-test-unit-main

**Pull request merged**